### DISCUSSION TITLE
How to troubleshoot slow server response

### DISCUSSION THREAD

**Discussion by jjb2016** [nextcloud/all-in-one#538] [Questions] [answered]
Hello all - hoping that somebody can help me here.  I have been running Nextcloud on TrueNas Core for a while now, and overall it has been great (apart from /usr/local/etc/nginx/conf.d/nextcloud.conf getting reset to default from time to time - but that's a separate issue).  My TrueNAS machine is only really meant to be a backup server and is a bit under-powered to run nextcloud I think.  So I decided to try and get Nextcloud installed on my main server, which is far more capable.  This is really a workstation machine running Arch Linux.  In the past I have had Nextcloud running on this machine as an LXC container, and as a KVM vertual machine, and I went through the pain of reading through various sources of information to get it all installed manually (on Arch Linux) ... this time I wanted to try a simpler deployment solution and so I've installed docker and the Nextcloud AIO container.  The install and setup process all went OK, without any apparent errors, but unfortunately nextcloud is very very slow to respond and load in my browser (or on my mobile app).  Sometimes it will be snappy as expected, but most times loading any nextcloud page takes ages.  Using my browsers developer tools I monitored some pages loads and in one instance simply opening the admin users settings page took more than 40 seconds.  Even just bringing up the sign in page takes a long time most times.

I should add that I've been setting up and testing this new nextcloud server while on the same LAN as the server, and that this slow response also occurs on the docker container host machine - so I don't believe this is network related.  Not sure if this is relevant, but I also configured docker to use the hosts ZFS storage.

Can anybody help me troubleshoot this?

**Comment by szaimen** [comment, marked as answer]
Just a few brain dump ideas:
update your containers to the latest available version from the aio interface, restart your containers from the aio interface, install server updates, restart your server. You could also try to disable all additions in an attempt to reduce server load and make Nextcloud as small/fast as possible 

If that does not work maybe ZFS is the problem as that was not tested by us...

Also you could try to use a local dns server like a pi-hole to point the domain in your network directly to the local ip-adress which is known to increase the network speed for Nextcloud in general by a lot.

**Reply by szaimen** [reply]
I've also recently seen this: https://github.com/nextcloud-snap/nextcloud-snap/issues/2058#issuecomment-1114028305
Maybe it helps resolving the issue...

**Reply by jjb2016** [reply]
Thank you for you response - and sorry for taking so long to reply.  I did actually resolve this issue.  It turned out to be a misconfiguration in my docker hosts firewall rules.  Thanks for the tip about using local DNS - I actually already have this setup and it's been working well for years.  I repurposed an old raspberry pi and have setup dnsmasq and unbound to effectively do the same job as Pi-Hole (did this before I discovered Pi-Hole!).  It definitely improves performance of nextcloud, and other self-hosted stuff, when accessed over the LAN.

**Reply by farzadha2** [reply]
hi @jjb2016 could you let me know  how did you fix this issue?
